### PR TITLE
Automate AM DTT and DTL.

### PR DESCRIPTION
**Pull request by ashercodeorg**
These serve as initial steps to further automating the DOTD script. Assuming positive feedback from DOTDs, further steps will be automated.

# TODO(asher): BEFORE REVIEW: Add URL feedback via the console to the user.

### CODE REVIEW

File: bin/dotd [hunk 1]
```diff
@@ -129,7 +130,9 @@ EOS
 user_name = ask_for_name
 
 should_i "morning DTT" do
-  open_pull_request 'test', 'staging', 'DTT (Staging > Test)'
+  GitHub.create_pull_request(
+    base: 'test', head: 'staging', title: 'DTT (Staging > Test)'
+  )
```
Comment (islemaster) [inline bin/dotd:135]: Key behavior change here: `open_pull_request` pops open a browser window to the pull request creation page with things autofilled, so the DOTD is two clicks from merging.  `GitHub.create_pull_request` takes care of the first click automatically (creating the pull request) but you're not auto-merging it here and I believe we're no longer opening the browser to make that easy to do.  So can we do one of:

* Use `open_url` to pop open the newly created pull request
* (My preference) Just auto-merge the PR here too

In fact, (and this is probably a future PR,) I'd love for the DOTD script here to also check the `#developers` topic and give you a heads-up if DTT is no for any reason (like you need to check on the robo-DTT).  Then it should give you a chance to loop back to this step once you've cleared any blocking issues :grin:.

Comment (ashercodeorg) [reply bin/dotd:135]: Though I agree that we want to auto-merge, I wanted to surface this before implementing it. In theory, DOTDs may be looking at the PR before merging it? Do we want to remove this ability?

File: bin/dotd [hunk 2]
```diff
@@ -129,7 +130,9 @@ EOS
 user_name = ask_for_name
 
 should_i "morning DTT" do
```
Comment (islemaster) [inline bin/dotd:132]: Add test right before this step that reminds you to check the robo-DTT results - or even detects and reports those results?

File: bin/dotd (hunk 1 above)
Comment (islemaster) [reply bin/dotd:135]: Might be worth polling; personally I don't evaluate a DTT PR for correctness before I merge anymore.  But I do see the argument that if we're removing guardrails like this, we should put better ones up (checking dev room topic).

If you'd like to proceed conservatively let's at least open the browser though.

**Comment by islemaster** [comment]
Any reason you're not replacing _all_ usages of `open_pull_request`, or just swapping out that helper's implementation?

**Comment by ashercodeorg** [comment]
In the interest of proceeding cautiously, I'm only swapping those steps that if executed accidentally, we would probably not care. Until things are better battle tested, for example, I'd rather not be auto-merging PRs to our `production` branch.

**Comment by ashercodeorg** [comment]
This is being closed in favor of PR#13692, which makes usage of additional helper methods in the `GitHub` module (added since the creation of this PR and in that PR).

**Pull request closed without merging**